proposals/FileSystemSubscriptionManager.md

```diff
@@ -0,0 +1,123 @@
+# Proposal: FileSystemSubscriptionManager
+
+## Authors
+
+* Ming-Ying Chung (<mych@google.com>)
+* Austin Sullivan (<asully@google.com>)
+
+## Background
+
+The [FileSystemObserver API][fso-api] allows browsing context to receive records of changes for the observed files or directories. However, changes which occur while the website has no open tabs are not visible to the website.
+
+This document proposes integrating file system change observing with service worker registration, such that a service worker may be woken up on a change to the local file system.
+
+[fso-api]: https://chromestatus.com/feature/4622243656630272
+
+## Goal
+
+Allow web applications that do not have an open tab to quickly respond to changes to registered files and directories on the local file system.
+
+## Discussion
+
+### Option 1. Enabling FileSystemObserver in Service Worker
+
+By [current design][fso], an instance of FileSystemObserver should only report changes which occur while the observer is connected and the website has an open & active tab.
+
+Enabling creations of FileSystemObserver in ServiceWorker will tie its lifetime with ServiceWorkerGlobalScope. Users might expect such observers to continue to watch files in the background. However, in browser implementation, service workers that haven’t received new events in a certain period of time, e.g. [30s in Chrome][chrome-sw], will likely be terminated.
+
+Hence, this approach might only work for websites that either can ensure a long-running service worker or are not really interested in using service worker registration for observing the file change events after the websites are closed.
+
+[fso]: https://docs.google.com/document/d/11o4HaslFwQxKIjJw0M9LAkkZp2ikCNwA9vvYsjA657w/edit?tab=t.0#heading=h.kktfoyrlhtnv
```

Comment:
All links should be to public docs. Let's link to this section of the original explainer? https://github.com/whatwg/fs/blob/main/proposals/FileSystemObserver.md#handling-changes-made-outside-the-lifetime-of-a-filesystemobserver

Reply:
Done.